enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/compiled/codegen/ir/expressions/Literal.scala

```diff
@@ -28,15 +28,22 @@ case class Literal(value: Object) extends CodeGenExpression {
   override def init[E](generator: MethodStructure[E])(implicit context: CodeGenContext) = {}
 
   override def generateExpression[E](structure: MethodStructure[E])(implicit context: CodeGenContext) = {
+    // When the literal value comes from the AST it should already have been converted
     val convertedValue = value match {
       case n: java.lang.Byte => n.longValue()
       case n: java.lang.Short => n.longValue()
-      case n: java.lang.Character => n.toLong
+      case n: java.lang.Character => n.toString
```

Comment:
Wouldn't this line change cause behavior changes?

Reply:
The previous behaviour is incorrect. A Java Character should map to a Cypher String. But it doesn't matter since we will never get here except if we wrote a unit tests for it. In https://github.com/neo4j/neo4j/pull/11378 I have replaced this conversion with an assert that it isn't needed instead.